lisa/commands.py

```diff
@@ -34,46 +36,46 @@ def _initialize(args: Namespace):
 
     # load external extension
     extends_config = config.getExtension()
-    _load_extends(extends_config)
+    _load_extends(config.base_path, extends_config)
```

Comment:
Do you mean external modules? Guess what we are doing today? People forks several repo and branches, change the code deeply. It have no chance to merge back.
The extension developers should fix side-effects problems themselves, the LISA keep platform clean and have some basic validation.
The loaded modules may break LISA, and it likes dynamic linked library. But extensibility is very important for LISA. We can see if any tool can help on some basic validations.

Reply:
This has nothing to do with forks of the repo. I mean that the `_load_extends` function does nothing but introduce side effects (note how we're calling it and not using its return value). This is a bad pattern because it means that the programmer must fully understand everything within `_load_extends` to be able to safely use it. This is where we want to learn from and use functional programming paradigms. _Some_ side effects can be okay, but they're exceptions which prove the rule. In general, functions should be side-effect free.

Reply:
Do you mean it should return a result, instead of manuplating global state? It looks load extends is an exception, it does operations on modules, and it triggers some operations on decorators, and furture calls on __subclasses__. It doesn't change other data can be returned.

Reply:
> Do you mean it should return a result, instead of manuplating global state?

Yes, manipulating state outside the scope of a function is a "[side effect](https://en.wikipedia.org/wiki/Side_effect_(computer_science))". We should design in such a way as to avoid as many side effects as possible.